internal/services/aws_reservation_service.go

```diff
@@ -35,23 +35,38 @@ func CreateAWSReservation(w http.ResponseWriter, r *http.Request) {
 	rDao := dao.GetReservationDao(r.Context())
 	pkDao := dao.GetPubkeyDao(r.Context())
 
-	// validate architecture match (hardcoded since image builder currently only supports x86_64)
-	supportedArch := "x86_64"
-	it := types.FindInstanceType(clients.InstanceTypeName(payload.InstanceType))
-	if it == nil {
-		renderError(w, r, payloads.NewInvalidRequestError(r.Context(), fmt.Sprintf("unknown type: %s", payload.InstanceType), UnknownInstanceTypeNameError))
+	// Either Launch Template or Instance Type must be set. Both can be set too, in that case, instance type overrides the launch template.
+	if payload.InstanceType == "" && payload.LaunchTemplateID == "" {
+		renderError(w, r, payloads.NewInvalidRequestError(r.Context(), "Both instance type and launch template are missing", BothTypeAndTemplateMissingError))
 		return
 	}
-	if it.Architecture.String() != supportedArch {
-		renderError(w, r, payloads.NewWrongArchitectureUserError(r.Context(), ArchitectureMismatch))
-		return
+
+	// Validate architecture match (hardcoded since image builder currently only supports x86_64). This can be only done
+	// when launch template is not set.
+	if payload.LaunchTemplateID == "" {
+		supportedArch := "x86_64"
+		it := types.FindInstanceType(clients.InstanceTypeName(payload.InstanceType))
+		if it == nil {
+			renderError(w, r, payloads.NewInvalidRequestError(r.Context(), fmt.Sprintf("unknown type: %s", payload.InstanceType), UnknownInstanceTypeNameError))
+			return
+		}
+		if it.Architecture.String() != supportedArch {
+			renderError(w, r, payloads.NewWrongArchitectureUserError(r.Context(), ArchitectureMismatch))
+			return
+		}
+	}
+
+	launchString := "Launch instance"
+	if payload.Amount > 1 {
+		launchString = "Launch instance(s)"
 	}
```

Comment:
Sneaking a small change - singulár vs plural as we know the amount. But this could actually be bad as we discussed during scrum today - these will be likely constants for i18n with descriptions. Therefore I think I will remove this.